tests/test_errors.py

```diff
@@ -107,6 +107,22 @@ def f(x, pred):
     assert jnp.isnan(y)
 
 
+def test_off_tracetime():
+    @jax.jit
+    def f(x):
+        return eqx.error_if(x, True, "hi", on_error="off")
```

Comment:
IIUC then I think XLA will just end up seeing this as the identity function. If so then note that XLA actually has special-casing for such functions – I don't think it matters here but I'd make this `error_if(...) + 1` just in case.